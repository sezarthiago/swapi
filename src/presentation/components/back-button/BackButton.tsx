import React from 'react';
import BackIcon from '../back-icon/BackIcon';
import Style from './BackButton.module.scss';

type Props = {
  onClick: () => void;
};

function BackButton({ onClick }: Props) {
  return (
    <button type="button" className={Style.backButton} onClick={onClick}>
      <BackIcon />
      <span>back</span>
    </button>
  );
}

export default BackButton;
