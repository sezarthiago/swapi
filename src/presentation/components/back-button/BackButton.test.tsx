import React from 'react';
import { render } from '@testing-library/react';
import BackButton from './BackButton';

describe('BackButton', () => {
  test('spapshot', () => {
    const sut = render(<BackButton onClick={() => {}} />);
    expect(sut).toMatchSnapshot();
  });
});
