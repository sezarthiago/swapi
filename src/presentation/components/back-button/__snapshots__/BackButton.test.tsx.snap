// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton spapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="backButton"
        type="button"
      >
        <svg
          fill="none"
          height="26"
          viewBox="0 0 31 26"
          width="31"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.0001 11.3333H6.69006L15.5117 2.51166L13.1551 0.154999L0.310059 13L13.1551 25.845L15.5117 23.4883L6.69006 14.6667H31.0001V11.3333Z"
            fill="white"
          />
        </svg>
        <span>
          back
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="backButton"
      type="button"
    >
      <svg
        fill="none"
        height="26"
        viewBox="0 0 31 26"
        width="31"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.0001 11.3333H6.69006L15.5117 2.51166L13.1551 0.154999L0.310059 13L13.1551 25.845L15.5117 23.4883L6.69006 14.6667H31.0001V11.3333Z"
          fill="white"
        />
      </svg>
      <span>
        back
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
